src/core/voteset.rs

```diff
@@ -198,8 +200,10 @@ impl VoteSet {
                             hash = h;
                         }
                         // inc the count of vote for hash
-                        if let Some(c) = votes_by_proposal.insert(hash, 1) {
-                            votes_by_proposal.insert(hash, c + 1);
+                        if let Some(c) = votes_by_proposal.get_mut(&hash) {
+                            *c += 1;
+                        } else {
+                            votes_by_proposal.insert(hash, 1);
```

Comment:
Better this way:
``` rust
*votes_by_proposal.entry(&hash).or_insert(0) += 1;
```
* [Rust std doc](https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html)
* [Rust Playground](https://play.rust-lang.org/?gist=9f0d7928e5ee745174483244cd64f90c&version=stable&mode=debug)